uaclient/apt.py

```diff
@@ -22,36 +23,37 @@
 CA_CERTIFICATES_FILE = '/usr/sbin/update-ca-certificates'
 
 
-class InvalidAPTCredentialsError(RuntimeError):
-    """Raised when invalid token is provided for APT access"""
-    pass
-
-
-def valid_apt_credentials(repo_url, username, password):
+def assert_valid_apt_credentials(repo_url, username, password):
```

Comment:
I felt like logic for raised exceptions should live inside valid_apt_credentials instead of returning True/False and then trying to parse apt-helper stderror content for 401s up in call sites. That prompted me to move the construction of the exception types inside this function.

Reply:
This could feasibly return `Tuple[bool, str]` (or even just `Optional[str]` with `None` indicating success), but as we don't have a consumer that _doesn't_ want the exceptions, I think what you're doing here is the correct choice.